src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -33,29 +43,59 @@ public final class PermissionEvaluationUtils {
             String.format(SERVICE_RESOURCE_FORMAT, SERVICE_PATTERN_STRING, SERVICE_RESOURCE_TYPE_PATTERN_STRING,
                     SERVICE_RESOURCE_NAME_PATTERN_STRING), Pattern.UNICODE_CHARACTER_CLASS);
 
+    private static final String POLICY_VARIABLE_FORMAT = "\\$\\{([a-z]+:[a-zA-Z.]+)\\}";
 
-    private PermissionEvaluationUtils() {
+    private static final Pattern POLICY_VARIABLE_PATTERN = Pattern.compile(POLICY_VARIABLE_FORMAT,
+            Pattern.CASE_INSENSITIVE);
+
+    private final GroupManager groupManager;
+
+    private static final Map<String, Pair<String,String>> policyVariableResolver =
+            new HashMap<String, Pair<String,String>>() {
+        {
+            put("iot:Connection.Thing.ThingName", new Pair<>("Thing", "ThingName"));
+        }
+    };
+
+    /**
+     * Constructor for PermissionEvaluationUtils.
+     *
+     * @param groupManager  Group Manager
+     */
+    @Inject
+    public PermissionEvaluationUtils(GroupManager groupManager) {
+        this.groupManager = groupManager;
     }
 
     /**
      * utility method of authorizing operation to resource.
      *
-     * @param operation             operation in the form of 'service:action'
-     * @param resource              resource in the form of 'service:resourceType:resourceName'
-     * @param groupToPermissionsMap device matching group to permissions map
-     * @return whether operation to resource in authorized
+     * @param request   Authorization Request
+     * @param session   Session
+     *
+     * @return boolean indicating if the operation requested is authorized
      */
-    public static boolean isAuthorized(String operation, String resource,
-                                       Map<String, Set<Permission>> groupToPermissionsMap) {
-        Operation op = parseOperation(operation);
-        Resource rsc = parseResource(resource);
+    public boolean isAuthorized(AuthorizationRequest request, Session session) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 93-126 into a separate method.